MedraPlugin.php

```diff
@@ -0,0 +1,394 @@
+                $contextDao->updateObject($context);
+            }
+        }
+    }
+
+    /**
+     * Provides submissions metadata in ONIX4DOI format for download
+     * @param Submission[] $submissions
+     */
+    public function exportSubmissions(array $submissions, Context $context): array
+    {
+        // Get filter and set objectsFileNamePart (see: PubObjectsExportPlugin::prepareAndExportPubObjects)
+        $exportPlugin = $this->_getExportPlugin();
+        $filterName = $exportPlugin->getSubmissionFilter();
+        $xmlErrors = [];
+
+        $temporaryFileId = $exportPlugin->exportAsDownload($context, $submissions, $filterName, 'articles', null, $xmlErrors);
+        return ['temporaryFileId' => $temporaryFileId, 'xmlErrors' => $xmlErrors];
+    }
+
+    /**
+     * Registers submissions DOIs
+     * @param Submission[] $submissions
+     */
+    public function depositSubmissions(array $submissions, Context $context): array
+    {
+        $exportPlugin = $this->_getExportPlugin();
+        $filterName = $exportPlugin->getSubmissionFilter();
+        $responseMessage = '';
+        $status = $exportPlugin->exportAndDeposit($context, $submissions, $filterName, $responseMessage);
+
+        return [
+            'hasErrors' => !$status,
+            'responseMessage' => $responseMessage
+        ];
+    }
+
+    /**
+     * Provides issues metadata in ONIX4DOI format for download
+     * @param Issue[] $issues
+     */
+    public function exportIssues(array $issues, Context $context): array
+    {
+        // Get filter and set objectsFileNamePart (see: PubObjectsExportPlugin::prepareAndExportPubObjects)
+        $exportPlugin = $this->_getExportPlugin();
+        $filterName = $exportPlugin->getIssueFilter();
+        $xmlErrors = [];
+
+        $temporaryFileId = $exportPlugin->exportAsDownload($context, $issues, $filterName, 'issues', null, $xmlErrors);
+        return ['temporaryFileId' => $temporaryFileId, 'xmlErrors' => $xmlErrors];
+    }
+
+    /**
+     * Registers issues DOIs
+     * @param Issue[] $issues
+     */
+    public function depositIssues(array $issues, Context $context): array
+    {
+        $exportPlugin = $this->_getExportPlugin();
+        $filterName = $exportPlugin->getIssueFilter();
+        $responseMessage = '';
+        $status = $exportPlugin->exportAndDeposit($context, $issues, $filterName, $responseMessage);
+
+        return [
+            'hasErrors' => !$status,
+            'responseMessage' => $responseMessage
+        ];
+    }
+
+    /**
+     * Add properties for mEDRA to the DOI entity for storage in the database.
+     *
+     * @param string $hookName Schema::get::doi
+     * @param array $args [
+     *
+     *      @option stdClass $schema
+     * ]
+     *
+     */
+    public function addToSchema(string $hookName, array $args): bool
+    {
+        $schema = &$args[0];
+
+        $settings = [
+            $this->getFailedMsgSettingName(),
```

Comment:
Are there any failed messages that would need to be carried forward as part of the migration? I didn't see anything about them there.

Reply:
Yes, there have been none till now -- we would only display the notification, but now, that we cannot easily display notifications, we save them.